x/marker/keeper/marker.go

```diff
@@ -562,21 +562,27 @@ func (k Keeper) TransferCoin(ctx sdk.Context, from, to, admin sdk.AccAddress, am
 	return nil
 }
 
-// SetMarkerMetadata updates the denom metadata records for the current marker.
-func (k Keeper) SetMarkerMetadata(ctx sdk.Context, metadata banktypes.Metadata, caller sdk.AccAddress) error {
+// SetMarkerDenomMetadata updates the denom metadata records for the current marker.
+func (k Keeper) SetMarkerDenomMetadata(ctx sdk.Context, metadata banktypes.Metadata, caller sdk.AccAddress) error {
 	if metadata.Base == "" {
 		return fmt.Errorf("invalid metadata request, base denom must match existing marker")
 	}
-	m, err := k.GetMarkerByDenom(ctx, metadata.Base)
-	if err != nil {
-		return fmt.Errorf("marker not found for %s: %w", metadata.Base, err)
+	marker, markerErr := k.GetMarkerByDenom(ctx, metadata.Base)
+	if markerErr != nil {
+		return fmt.Errorf("marker not found for %s: %w", metadata.Base, markerErr)
 	}
-	if !m.GetManager().Equals(caller) && !m.AddressHasAccess(caller, types.Access_Admin) {
+	if !marker.GetManager().Equals(caller) && !marker.AddressHasAccess(caller, types.Access_Admin) {
```

Comment:
manager is only set on proposed/finalized markers ... this value is cleared and the permissions list takes over for active markers this check should be structured as an OR not an AND

Reply:
The only thing changed on this line is the variable name `m` was changed to `marker` because it was being used several lines down.

I thought about switching it to an or as well, but then I noticed that the equals are negated. So this line as it is returns an error if the caller is neither the manager nor an admin.

Reply:
Ah... serves me right for reviewing in a hurry.